oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,140 @@
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.jose.jws.JwsAlgorithms;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtVerifier;
+
+import javax.crypto.SecretKey;
+import java.security.PublicKey;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+public class OAuth2ResourceServerConfigurer<B extends HttpSecurityBuilder<B>> extends
+		AbstractHttpConfigurer<JwtVerifierConfigurer<B>, B> {
+
+	public BearerConfigurer bearer() {}
+
+	public class BearerConfigurer {
+		public BearerConfigurer inBody() {}
+
+		public BearerConfigurer inHeader() {}
+
+		public BearerConfigurer inQuery() {}
+
+		public OAuth2ResourceServerConfigurer<B> and() {}
+	}
+
+
+
+	public JwtConfigurer jwt() {}
```

Comment:
The DSL should not depend on`JwtDecoder`. The only configuration required for the initial feature implementation is JWK Set Url. I don't think anything more than that is required at the moment.

Reply:
What are you thoughts about the code needing to provide a custom `JwtDecoder` implementation?

Reply:
An `AuthenticationProvider` may be configured by the user and associate it with a custom `JwtDecoder`. The `AuthenticationProvider` can than be registered as a `@Bean`.

Or we can allow the user to register a `JwtDecoder` as a `@Bean` and than the `SecurityConfigurer` can discover it and supply it to the `JwtAuthenticationProvider`.

There are likely other options we can leverage without coupling it to the DSL.